ecs-agent/tmds/handlers/taskprotection/v1/handlers/handlers_test.go

```diff
@@ -0,0 +1,671 @@
+
+	if tc.setAgentStateExpectations != nil {
+		tc.setAgentStateExpectations(agentState)
+	}
+	if tc.setCredsManagerExpectations != nil {
+		tc.setCredsManagerExpectations(credsManager)
+	}
+	if tc.setFactoryExpectations != nil {
+		tc.setFactoryExpectations(ctrl, factory)
+	}
+	if tc.setMetricsExpectations != nil {
+		tc.setMetricsExpectations(ctrl, metricsFactory)
+	}
+
+	// Setup the handlers
+	router := mux.NewRouter()
+	router.HandleFunc(
+		TaskProtectionPath(),
+		GetTaskProtectionHandler(agentState, credsManager, factory, cluster, metricsFactory, ecsCallTimeout),
+	).Methods("GET")
+	router.HandleFunc(
+		TaskProtectionPath(),
+		UpdateTaskProtectionHandler(agentState, credsManager, factory, cluster, metricsFactory, ecsCallTimeout),
+	).Methods("PUT")
+
+	// Create the request
+	method := "GET"
+	var requestBody io.Reader
+	if tc.requestBody != nil {
+		method = "PUT"
+		reqBodyBytes, err := json.Marshal(tc.requestBody)
+		require.NoError(t, err)
+		requestBody = bytes.NewReader(reqBodyBytes)
+	}
+	req, err := http.NewRequest(method, fmt.Sprintf("/api/%s/task-protection/v1/state", endpointId),
+		requestBody)
+	require.NoError(t, err)
+
+	// Send the request and record the response
+	recorder := httptest.NewRecorder()
+	router.ServeHTTP(recorder, req)
+
+	// Parse the response body
+	var actualResponseBody types.TaskProtectionResponse
+	err = json.Unmarshal(recorder.Body.Bytes(), &actualResponseBody)
+	require.NoError(t, err)
+
+	// Assert status code and body
+	assert.Equal(t, tc.expectedStatusCode, recorder.Code)
+	assert.Equal(t, tc.expectedResponseBody, actualResponseBody)
+
+	// Run any post assertions
+	if tc.postAssertions != nil {
+		tc.postAssertions(t)
+	}
+}
+
+func TestGetTaskProtection(t *testing.T) {
+	// Initialize some data common to the test cases
+	happyECSInput := ecs.GetTaskProtectionInput{
+		Cluster: aws.String(cluster),
+		Tasks:   aws.StringSlice([]string{taskARN}),
+	}
+	metricName := metrics.GetTaskProtectionMetricName
+
+	// A helper function for setting expectations on mock ECS Client Factory
+	factoryExpectations := func(
+		input ecs.GetTaskProtectionInput,
+		output *ecs.GetTaskProtectionOutput,
+		err error,
+	) func(*gomock.Controller, *MockTaskProtectionClientFactoryInterface) {
+		return func(ctrl *gomock.Controller, factory *MockTaskProtectionClientFactoryInterface) {
+			client := mock_api.NewMockECSTaskProtectionSDK(ctrl)
+			client.EXPECT().GetTaskProtectionWithContext(gomock.Any(), &input).Return(output, err)
+			factory.EXPECT().NewTaskProtectionClient(taskRoleCreds()).Return(client)
+		}
+	}
+
+	// Test cases start here
+	t.Run("task lookup failure", func(t *testing.T) {
+		testTaskProtectionRequest(t, taskMetadataLookupFailureCase(metricName, nil))
+	})
+	t.Run("task metadata fetch failure", func(t *testing.T) {
+		testTaskProtectionRequest(t, taskMetadataFetchErrorCase(
+			state.NewErrorMetadataFetchFailure(""), metricName, nil))
+	})
+	t.Run("task metadata uknown error", func(t *testing.T) {
```

Comment:
nit: typo `uknown`

Reply:
Good catch. Let me submit a follow-up PR for this as functional tests have been super flaky lately and as the typo is in a test case name.